openage/convert/terrain_convert.pyx

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright 2018-2018 the openage authors. See copying.md for legal info.
+
+"""
+Transforms the dimetric tiles converted from SLP files to cartesian format.
+"""
+
+import math
+import argparse
+from PIL import Image
+from .slp import SLP
+from .texture import Texture
+
+class Terrain_convert:
+
+    def merge(img):
```

Comment:
For efficiency, the merge and transformation should be written as cdef methods and operate on an ndarray instead of pillow images. You can achieve that by defining and writing `merge` and `transform` as cdef methods and call them from a normal Python def method. The def method will be callable from "the outside", i.e. other Python modules. See `slp.pyx`. You can see how it works there with `get_picture_data` and `determine_damage_matrix`.

You also want as much C logic as possible for maximum speed.